src/scrapers/movie/TMDb.cpp

```diff
@@ -845,12 +846,19 @@ void TMDb::parseAndAssignInfos(QString json, Movie* movie, QVector<MovieScraperI
 
     // Trailers
     if (infos.contains(MovieScraperInfos::Trailer) && parsedJson.value("youtube").isArray()) {
-        // The trailer listed first is most likely also the best.
-        const auto firstTrailer = parsedJson.value("youtube").toArray().first().toObject();
-        if (!firstTrailer.value("source").toString().isEmpty()) {
-            const QString youtubeSrc = firstTrailer.value("source").toString();
-            movie->setTrailer(
-                QUrl(helper::formatTrailerUrl(QStringLiteral("https://www.youtube.com/watch?v=%1").arg(youtubeSrc))));
+        // Look for "type" key in each element and look for the first instance of "Trailer" as value
+        const auto videos = parsedJson.value("youtube").toArray();
+        for (const auto& it : videos) {
+            const auto videoObj = it.toObject();
+            const QString videoType = videoObj.value("type").toString();
+            if (videoType == "Trailer") {
+                const QString youtubeSrc = videoObj.value("source").toString();
+                movie->setTrailer(QUrl(
+                    helper::formatTrailerUrl(QStringLiteral("https://www.youtube.com/watch?v=%1").arg(youtubeSrc))));
+                break;
+            } else {
+                continue;
```

Comment:
Out of curiosity: What other than a trailer could there be? Wait.. Are teasers available as well? :)

Reply:
Nevermind. You explained this already... :)